File: app/src/main/java/vgame/ir/data/local/entity/ChapterEntity.java
package vgame.ir.data.local.entity;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

import java.util.List;

@Entity(tableName = "chapter")
public class ChapterEntity {

    @PrimaryKey
    @SerializedName("id")
    private int id;

    @SerializedName("title")
    private String title;

    @SerializedName("course_id")
    private int course_id;

    @Ignore
    @SerializedName("sessions")
    private List<LessonEntity> lessons;

    @Ignore
    @SerializedName("course")
    private CourseEntity course;

    public CourseEntity getCourse() {
        return course;
    }

    public void setCourse(CourseEntity course) {
        this.course = course;
    }

    public int getCourse_id() {
        return course_id;
    }

    public void setCourse_id(int course_id) {
        this.course_id = course_id;
    }

    public String getTitle() {
        return title;
    }

    public void getTitle(String name) {
        this.title = name;
    }

    public List<LessonEntity> getLessons() {
        return lessons;
    }

    public void setLessons(List<LessonEntity> lessons) {
        this.lessons = lessons;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setTitle(String title) {
        this.title = title;
    }
}

File: app/src/main/java/vgame/ir/data/local/entity/ChapterEntity.kt
package vgame.ir.data.local.entity

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

import com.google.gson.annotations.SerializedName

@Entity(tableName = "chapter")
class ChapterEntity {

    @PrimaryKey
    @SerializedName("id")
    var id: Int = 0

    @SerializedName("title")
    var title: String? = null

    @SerializedName("course_id")
    var course_id: Int = 0

    @Ignore
    @SerializedName("sessions")
    var lessons: List<LessonEntity>? = null

    @Ignore
    @SerializedName("course")
    var course: CourseEntity? = null

    fun getTitle(name: String) {
        this.title = name
    }
}
